### PR TITLE
Upgrade for terraform 0.12

### PR DESCRIPTION
**Pull request by OGProgrammer**
My scripts are blocked because this module isn't updated...

### CODE REVIEW

**Comment by OGProgrammer** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here (e.g. `I signed it!`) and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google.com/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2FGoogleCloudPlatform%2Fterraform-google-sql-db%2Fpull%2F36) for more info**.

I signed it!

File: main.tf [hunk 1]
```diff
@@ -14,51 +14,145 @@
...
+        hour         = lookup(maintenance_window.value, "hour", null)
+        update_track = lookup(maintenance_window.value, "update_track", null)
+      }
+    }
+    disk_size        = var.disk_size
+    disk_type        = var.disk_type
+    pricing_plan     = var.pricing_plan
+    replication_type = var.replication_type
+        // This will require more checking if database_flags was set before we declare the dynamic block
+//    dynamic "database_flags" {
```
Comment (OGProgrammer) [inline main.tf]: Before you merge this, this needs to be updated.

**Comment by namusyaka** [comment]
@OGProgrammer This is amazing!
But I'm planning to merge #35 before merging this.
Could you work after that?

**Comment by OGProgrammer** [comment]
I see its merged so I'll see what I can do about fixing this up for merging

**Comment by namusyaka** [comment]
@OGProgrammer Yes, please.

**Comment by rosera** [comment]
In the variable.tf - Line 125. Not sure if this is an error, assuming the line should read default = {} and not default =[{}]?

**Comment by ingwarr** [comment]
I'm working on it now https://github.com/GoogleCloudPlatform/terraform-google-sql-db/pull/53

**Comment by OGProgrammer** [comment]
Merge 53 !

**Pull request closed without merging**